extras/mmu/mmu.py

```diff
@@ -2132,6 +2132,16 @@ def _calibrate_bowden_length_auto(self, approximate_length, extruder_homing_max,
                 self._initialize_filament_position(dwell=True)
                 self._load_gate(allow_retry=False)
                 self._load_bowden(approximate_length)
+
+                if self.extruder_homing_endstop == self.ENDSTOP_EXTRUDER_ENTRY and self.sensor_manager.check_sensor(self.ENDSTOP_EXTRUDER_ENTRY):
+                    msg = "Fast move overshoot. The %s sensor triggered before homing. Try reducing your estimated BOWDEN_LENGTH." % self.extruder_homing_endstop
+                    self.log_error(msg)
+                    # Home back to the gate
+                    self._wrap_gcode_command(self.pre_unload_macro, exception=True, wait=True)
+                    self._unload_gate(approximate_length)
+                    self._wrap_gcode_command(self.post_unload_macro, exception=True, wait=True)
+                    raise MmuError("Fast move overshoot. The %s sensor triggered before homing. Try reducing your estimated BOWDEN_LENGTH." % self.extruder_homing_endstop)
```

Comment:
This prevents "successful" bowden calibration when you give it too large of an estimated bowden length. What would happen if you provided too large of an estimated length is it could collide with the extruder and slip on the filament. Then when it does the `_home_to_extruder` it would see the sensor is already triggered so it doesn't do any additional homing and we end up with an "calibrated" length that is too long. These changes will detect that sensor triggering before it was supposed to and instructs the user to try again with a shorter bowden length.